src/rust/container/lib.rs

```diff
@@ -0,0 +1,88 @@
+use bollard::API_DEFAULT_VERSION;
+use bollard::Docker;
+use bollard::exec::CreateExecOptions;
+use bollard::exec::StartExecOptions;
+use bollard::query_parameters::StopContainerOptionsBuilder;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum ContainerError {
+    #[error("Connection error: {0}")]
+    ConnectionError(String),
+    #[error("Container error: {0}")]
+    DestroyError(String),
+    #[error("Container error: {0}")]
+    ExecError(String),
+}
+
+#[cxx::bridge(namespace = "workerd::rust::container")]
+mod ffi {
+    extern "Rust" {
+        pub async fn start(
+            container_name: &str,
+            entrypoint: Vec<&str>,
+            env: Vec<&str>,
+        ) -> Result<String>;
+        pub async fn destroy(container_name: &str) -> Result<()>;
+        pub async fn signal(container_name: &str, signal: &str) -> Result<()>;
+    }
+}
+
+// TODO:
+// [ ] - kj::Promise<void> status(StatusContext context) override;
+// [x] - kj::Promise<void> start(StartContext context) override;
+// [ ] - kj::Promise<void> monitor(MonitorContext context) override;
+// [x] - kj::Promise<void> destroy(DestroyContext context) override;
+// [x] - kj::Promise<void> signal(SignalContext context) override;
+// [ ] - kj::Promise<void> getTcpPort(GetTcpPortContext context) override;
+// [ ] - kj::Promise<void> listenTcp(ListenTcpContext context) override;
+
+pub fn connect() -> Result<Docker, ContainerError> {
+    let addr = "/var/run/docker.sock";
+    let socket = Docker::connect_with_unix(addr, 30, API_DEFAULT_VERSION)
```

Comment:
It looks like using [connect_with_socket](https://docs.rs/bollard/latest/bollard/struct.Docker.html#method.connect_with_socket) would support both windows and linux?

Reply:
Docker class comes from the third party library.